test/mpi/maint/generate_gpu_tests.sh

```diff
@@ -0,0 +1,61 @@
+#! /bin/sh
+##
+## Copyright (C) by Argonne National Laboratory
+##     See COPYRIGHT in top-level directory
+##
+
+# Generate testlist.gpu for cases not covered under gentests-dtop.sh
+# and test_coll_algos.sh
+# Reads input from the input file in format "testpath/testname:procs"
+# and creates test combinations for (host, device) and (device, device).
+
+srcdir=`dirname $0`
+builddir=$PWD
+
+while read -r line ; do
+    if [ ! `echo $line | cut -c 1` = "#" ] ; then
+        # the line is not a comment
+        pathname=`echo $line | cut -f1 -d':'`
+        procs=`echo $line | cut -f2 -d':'`
+        other_args=""
+    else
+        # the line is a comment
+        continue
+    fi
+
+    if [ -z "$pathname" ] ; then
+        echo "No pathname found"
+        exit 1
+    elif [ -z "$procs" ] ; then
+        echo "$pathname has no proc count specified"
+        exit 1
+    fi
+
+    # get test directory and name
+    testdir=`dirname $pathname`
+    testname=`basename $pathname`
+
+    # if this is the first test in this directory, create a new
+    # testlist.gpu
+    found=0
+    for dir in $dirs ; do
+        if test "$testdir" = "$dir" ; then
+            found=1
+        fi
+    done
+    if test "$found" = "0" ; then
+        dirs="$dirs $testdir"
+        printf "" >> ${builddir}/${testdir}/testlist.gpu
```

Comment:
```suggestion
        printf "" > ${builddir}/${testdir}/testlist.gpu
```
I ran configure several times when doing manual tests, which resulted in a testlist.gpu file with many duplicate entries. Should we make sure to overwrite the file each time this test is run?

Reply:
The `testlist.gpu` is first created by `gentests_dtp.sh` so my intent was to append to it. While rebasing today, I noticed that I had accidently removed the similar line of code from `gentests_dtp.sh`, which would make this line append over and over again. I think the current case should work but I will verify it again.

Reply:
OK, great.

Reply:
I verified this and its working as expected.